docs/fundamentals/gateway/custom-gateway.md

```diff
@@ -0,0 +1,328 @@
+(custom-gateway)=
+# Customize the Gateway
+A Jina Gateway is customizable and can be implemented in the same way Executors are implemented.
+With Custom Gateways, Jina gives power back to the users, by allowing to implement any server, protocol and interface on
+the gateway level. This means you have more freedom in:
+* Choosing your favourite server framework.
+* Choosing the protocol used to serve your app.
+* Defining your API Gateway interface. You can define your JSON schema or protos,...
+
+Customization is allowed different components:
+* Implementing the custom gateway using a `base` gateway class: {class}`~jina.Gateway` or {class}`~jina.serve.runtimes.gateway.http.fastapi.FastAPIBaseGateway`.
```

Comment:
Separate the 2 options. Let's focus on explaining BaseGateway and then have another section for FastAPI.

BTW. For FastAPI we could already expose the health check?

Reply:
There's already a section called `Implementing the custom gateway` which is split in 2 parts for each base gateway